pkg/scheduler/scheduler.go

```diff
@@ -398,22 +395,32 @@ func (s *Scheduler) doScheduleBinding(namespace, name string) error {
 		return err
 	}
 
+	// Update "Scheduled" condition according to schedule result.
+	defer func() {
+		var condition metav1.Condition
+		if err == nil {
+			condition = util.NewCondition(workv1alpha2.Scheduled, scheduleSuccessReason, scheduleSuccessMessage, metav1.ConditionTrue)
+		} else {
+			condition = util.NewCondition(workv1alpha2.Scheduled, scheduleFailedReason, err.Error(), metav1.ConditionFalse)
+		}
+		if updateErr := s.updateBindingScheduledConditionIfNeeded(rb, condition); updateErr != nil {
+			klog.Errorf("Failed update condition(%s) for ResourceBinding(%s/%s)", workv1alpha2.Scheduled, rb.Namespace, rb.Name)
+			if err == nil {
+				// schedule succeed but update condition failed, return err in order to retry in next loop.
+				err = updateErr
+			}
+		}
+	}()
+
 	start := time.Now()
-	if !helper.IsBindingReady(&rb.Status) {
-		// the binding has not been scheduled, need schedule
-		klog.Infof("Start scheduling ResourceBinding(%s/%s)", namespace, name)
-		err = s.scheduleResourceBinding(rb)
-		metrics.BindingSchedule(string(FirstSchedule), metrics.SinceInSeconds(start), err)
```

Comment:
@qianjun1993 
Here merged `FirstSchedule` to `ReconcileSchedule` because they share the same schedule logic, are you ok with it?

Reply:
I am ok with that for they share the same schedule logic now. But As mentioned in #900, there may be some differences in the two kind of schedule.